packages/ai-core/src/common/language-model.ts

```diff
@@ -32,7 +32,14 @@ export const isLanguageModelRequestMessage = (obj: unknown): obj is LanguageMode
         'query' in obj &&
         typeof (obj as { query: unknown }).query === 'string'
     );
-export type ToolRequestParametersProperties = Record<string, { type: string, [key: string]: unknown }>;
+
+export interface ToolRequestParameterProperty {
+    type?: string;
+    anyOf?: ToolRequestParameterProperty[];
```

Comment:
As said in another comment: `oneOf` and `allOf` are other commonly used constructs in JSON Schema